### DISCUSSION TITLE
Get Realtime Connection from Hook

### DISCUSSION THREAD

**Discussion by RickEyre** [feathersjs/feathers#2224] [Q&A] [answered]
I'd like to subscribe users to channels after they initially do a `get` or `find` request:

```js
app.service('myservice').hooks({
  find: (context) {
    app.channel(`/rooms/${context.params.id}`).join(connection)
  }
}
```

I haven't seen any documentation on how to retrieve the `connection` object in hooks. Is this possible? If it's not possible, is there a solution to this general problem?

**Comment by J3m5** [comment, marked as answer]
The connection can be found in `context.params.connection`.
See the [docs there](https://docs.feathersjs.com/api/services.html#params)

**Reply by RickEyre** [reply]
Awesome, thank you.
